src/pipeline/eval.py

```diff
@@ -16,46 +17,26 @@ def _get_eval_prompt_tmpl(eval_prompt_tmpl_path):
     prompts = toml.load(eval_prompt_tmpl_path)
     return prompts['eval']['prompt']
 
-def _construct_eval_prompts(batch_data, lm_responses, eval_prompt_tmpl):
-    """
-    construct_eval_prompt returns a prompt to be injected into the language model (evaluator)
-
-    arguments:
-    ds -- a batch data records which has "prompt", "messages" columns
-    lm_response -- string value which fine-tuned model generated
-    eval_prompt_tmpl -- string with placeholders of instruction, human_response, and lm_response.
-    """
-    eval_prompts = []
-    for (instruction, messages, lm_response) in zip(
-        batch_data['prompt'], batch_data['messages'], lm_responses
-    ):
-        ground_truth = messages[1]['content']
-        
-        eval_prompts.append(
-            Template(eval_prompt_tmpl).substitute(
-                instruction=instruction,
-                human_response=ground_truth,
-                lm_response=lm_response
-            )         
-        )
-
-    return eval_prompts
-    
-def _get_test_dataset(dataset_id, split, tokenizer, batch_size):
+def _get_lm_response_dataset(dataset_id, eval_prompt_tmpl, batch_size):
     """
     _get_test_dataset returns the target test dataset
     """
     def __batch_process(batch):
-        batch["input_ids"] = [
-            tokenizer.apply_chat_template(
-                [{"role": "user", "content": prompt}],
-                add_generation_prompt=True, tokenize=False
+        eval_prompts = []
+        for (instruction, target_response, candidate_response) in zip(
+            batch['instructions'], batch['target_responses'], batch['candidate_responses']
+        ):
+            eval_prompts.append(
+                Template(eval_prompt_tmpl).substitute(
+                    instruction=instruction,
+                    human_response=target_response,
+                    lm_response=candidate_response
+                )         
             )
-            for prompt in batch["prompt"]
-        ]
+        batch["eval_prompts"] = eval_prompts
         return batch
         
-    return load_dataset(dataset_id, split=split).map(
+    return load_dataset(dataset_id, split='train').map(
```

Comment:
Why are we generating responses from the "train" set for evaluation? Also, let's make this split name configurable.

Reply:
make sense! 
actually, I don't have a solid background on this. 
I thought `train` is the default split for any Dataset.

Reply:
No. The robots dataset doesn’t have that split.

Reply:
if I don't specify anything about split, `train` becomes default split. 
(i.e. https://huggingface.co/datasets/chansung/eval_dataset_test)

but yeah, make it configurable makes sense!